### DISCUSSION TITLE
XML source not synchronized with outline view

### DISCUSSION THREAD

**Discussion by MotasemSH** [eclipse-birt/birt#1849] [General]
Hi all
I am developing a BIRT report, when an error shows up that something is wrong with the report I search for it inside "XML Source", the error is indicated with red mark on the side of xml source, if I clicked on the error spot in xml source the outline view is not synchronized with the same position.
Also there is no way to search for specific element id inside outline, if there were a search item to filter the view for some text.

Is there any solution or preference setting?

**Comment by merks** [comment]
Someone else will need to reproduce what you’re seeing.  Does the outline change based on selecting text in the source?

**Reply by MotasemSH** [reply]
No, but when choosing any element in outline the source is navigating to the related position.

**Reply by merks** [reply]
If you change the cursor location in the source, does the outline change to match?

**Reply by MotasemSH** [reply]
No

**Comment by hvbtup** [comment]
> Is there any solution or preference setting?

There is no preference setting.

Anyway, I like your idea to allow searching the outline view for an element ID for two reasons:

1) Sometimes, you get a stack trace containing an ID at runtime, not in the BIRT designer.

2) While it is already possible to search for an ID in the XML, this isn't very helpful in case of a really complex report (in my experience).

What I do in such cases is to search the ID in the XML and then search "upwards left" to find at least a container element which contains the item to search for. And I know the type of the element with the ID (e.g. a text item). Then I click through the outline, watching the ID in the property view to find the element. This is cumbersome, but I don't think there is a better way till now.

Of course it helps if you *name* your items, at least the container items (grids, list, tables).

**Reply by MotasemSH** [reply]
Thank you for replying, 
I appreciate your suggestion, unfortunately when the report is complex it is frustrating to navigate through outline to find an element with specific ID.
